src/VisualStudio/TestUtilities/VisualStudioInstance.cs

```diff
@@ -20,42 +18,58 @@ public class VisualStudioInstance
     {
         private readonly Process _hostProcess;
         private readonly DTE _dte;
-        private readonly IntegrationService _service;
-        private readonly string _serviceUri;
-        private readonly IpcClientChannel _serviceChannel;
+        private readonly IntegrationService _integrationService;
+        private readonly IpcClientChannel _integrationServiceChannel;
 
         // TODO: We could probably expose all the windows/services/features of the host process in a better manner
-        private readonly Lazy<InteractiveWindow> _csharpInteractiveWindow;
+        private readonly Lazy<CSharpInteractiveWindow> _csharpInteractiveWindow;
         private readonly Lazy<EditorWindow> _editorWindow;
         private readonly Lazy<SolutionExplorer> _solutionExplorer;
+        private readonly Lazy<Workspace> _workspace;
 
         public VisualStudioInstance(Process process, DTE dte)
         {
             _hostProcess = process;
             _dte = dte;
 
-            ExecuteDteCommandAsync("Tools.StartIntegrationTestService").GetAwaiter().GetResult();
+            dte.ExecuteCommandAsync(VisualStudioCommandNames.VsStartServiceCommand).GetAwaiter().GetResult();
 
-            _serviceChannel = new IpcClientChannel($"ipc channel client for {_hostProcess.Id}", sinkProvider: null);
-            ChannelServices.RegisterChannel(_serviceChannel, ensureSecurity: true);
+            _integrationServiceChannel = new IpcClientChannel($"IPC channel client for {_hostProcess.Id}", sinkProvider: null);
+            ChannelServices.RegisterChannel(_integrationServiceChannel, ensureSecurity: true);
 
             // Connect to a 'well defined, shouldn't conflict' IPC channel
-            _serviceUri = string.Format($"ipc://{IntegrationService.PortNameFormatString}", _hostProcess.Id);
-            _service = (IntegrationService)(Activator.GetObject(typeof(IntegrationService), $"{_serviceUri}/{typeof(IntegrationService).FullName}"));
+            var serviceUri = string.Format($"ipc://{IntegrationService.PortNameFormatString}", _hostProcess.Id);
+            _integrationService = (IntegrationService)(Activator.GetObject(typeof(IntegrationService), $"{serviceUri}/{typeof(IntegrationService).FullName}"));
+            _integrationService.Uri = serviceUri;
 
-            _csharpInteractiveWindow = new Lazy<InteractiveWindow>(() => InteractiveWindow.CreateCSharpInteractiveWindow(this));
+            // There is a lot of VS initialization code that goes on, so we want to wait for that to 'settle' before
+            // we start executing any actual code.
+            _integrationService.Execute(typeof(RemotingHelper), nameof(RemotingHelper.WaitForSystemIdle));
```

Comment:
Why is this sufficient? How do you know something you rely on isn't on a timer that will wait a second or two?